modules/security/src/main/java/org/apache/ignite/internal/security/authentication/AuthenticationProviderEqualityVerifier.java

```diff
@@ -61,6 +64,21 @@ public static boolean areEqual(@Nullable AuthenticationProviderView o1, @Nullabl
     }
 
     private static boolean areEqual(BasicAuthenticationProviderView o1, BasicAuthenticationProviderView o2) {
-        return o1.username().equals(o2.username()) && o1.password().equals(o2.password());
+        NamedListView<? extends BasicUserView> users1 = o1.users();
+        NamedListView<? extends BasicUserView> users2 = o2.users();
+        if (users1.size() != users2.size()) {
+            return false;
+        }
+
+        for (BasicUserView basicUserView : users1) {
+            BasicUserView basicUser2View = users2.get(basicUserView.username());
+            if (basicUser2View == null) {
+                return false;
+            }
+            return Objects.equals(basicUserView.username(), basicUser2View.username())
```

Comment:
You do an early return from the cycle. So, if the size of the collections is equal and the first users are equal you return true. But the rest of the users can be different. I think here you want to continue iteration or return false if you found the unmatched.